source/ada/lsp-ada_handlers.adb

```diff
@@ -403,7 +409,12 @@ package body LSP.Ada_Handlers is
          end if;
       end if;
 
-      Self.Context.Load_Document (Value.textDocument);
+      Document := Self.Context.Load_Document (Value.textDocument);
+
+      Document.Get_Errors (Diag.diagnostics);
```

Comment:
(does Document need to be freed?)

Reply:
No. All documents are managed by Context. I've added comments and disabled document deallocation outside of context.